nomad/job_endpoint_statuses.go

```diff
@@ -200,6 +201,16 @@ func (j *Job) Statuses(
 func jobStatusesJobFromJob(ws memdb.WatchSet, store *state.StateStore, job *structs.Job) (structs.JobStatusesJob, uint64, error) {
 	highestIdx := job.ModifyIndex
 
+	isPack := false
+	if job.Meta != nil {
+		for key, value := range job.Meta {
+			if strings.HasPrefix(key, "pack") && value != "" {
+				isPack = true
+				break
+			}
+		}
+	}
+
```

Comment:
This could very well be too expensive for this endpoint, but I'm a little blind to how much we ought to tolerate here. For example, the possibility of a job with many, many instances of metadata entries gives me pause. If there is a more clear-cut indicator of "this came from pack", I'd love to use that, but I think metadata is the best bet.

Reply:
since we know specific `pack.` meta key(s), instead of iterating the map which could potentially be arbitrarily long, as you mentioned, and getting string processing involved, I think we can do a simple map lookup.  also fun tidbit: you can attempt to access a nil map without nil-checking it, so I think all this can boil down to:

```go
	_, isPack := job.Meta["pack.name"]
```

but also, and this is totally personal taste, I like to do static type definition of what I'm gonna return up top, then populate it with whatever extra processing afterwards.  so I'd personally do this down underneath `jsj` definition somewhere instead:

```go
	_, jsj.IsPack = job.Meta["pack.name"]
```

Reply:
This is way simpler and setting it unorphaned makes a lot of sense. Updating now!